pkg/operator/controller/names.go

```diff
@@ -20,19 +20,63 @@ import (
 	"fmt"
 	"hash"
 	"hash/fnv"
+	"strings"
 
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/apimachinery/pkg/util/rand"
+	"k8s.io/client-go/discovery"
 
 	operatorv1alpha1 "github.com/openshift/external-dns-operator/api/v1alpha1"
 	operatorconfig "github.com/openshift/external-dns-operator/pkg/operator/config"
 )
 
 const (
 	// ExternalDNSBaseName is the base name for any ExternalDNS resource.
-	ExternalDNSBaseName = "external-dns"
+	ExternalDNSBaseName         = "external-dns"
+	CredentialsRequestNamespace = "openshift-cloud-credential-operator"
+	kind                        = "OpenShiftAPIServer"
+	group                       = "operator.openshift.io"
+	version                     = "v1"
 )
 
+// Returns true if platform is OCP
+func IsOCP(kubeClient discovery.DiscoveryInterface) bool {
```

Comment:
yes but do you know a common place where I can add it so that the webhook file also can access it?

Reply:
Since the function in `externaldnscontroller` is already exported it can be used in the `api` package.

Reply:
I did try that but gives a cyclic dependency error.

Reply:
Ignore the above ^^ I got your point

Reply:
go run sigs.k8s.io/controller-tools/cmd/controller-gen "crd:trivialVersions=true,preserveUnknownFields=false" rbac:roleName=external-dns-operator webhook paths="./..." output:crd:artifacts:config=config/crd/bases
-: import cycle not allowed: import stack: [github.com/openshift/external-dns-operator/api/v1alpha1 github.com/openshift/external-dns-operator/pkg/operator/controller github.com/openshift/external-dns-operator/api/v1alpha1]
/home/miheer/.gvm/gos/go1.16.3/src/github.com/openshift/external-dns-operator/api/v1alpha1/externaldns_webhook.go:22:24: no such package located
Error: not all generators ran successfully

Reply:
I defined the function in the webhook so not it is accessible.

Reply:
Personally I would like to see this refactored completed. We don't need to verify if the cluster is an OCP cluster every time. Once at startup is sufficient and reasonable. So that bool could be passed as a parameter to both the webhook and the controller.

Reply:
done

Reply:
I agree with Arjun on the point that the check is sufficient to be done once at startup.
This would also simplify the other parts of the code: for instance, Kube client's `DiscoveryInterface` won't need to be used in the controllers.